File: TransCoder_tokenized_test_set_functions/java/STACK_PERMUTATIONS_CHECK_IF_AN_ARRAY_IS_STACK_PERMUTATION_OF_OTHER.java
static boolean checkStackPermutation ( int ip [ ], int op [ ], int n ) {
  Queue < Integer > input = new LinkedList < > ( );
  for ( int i = 0;
  i < n;
  i ++ ) {
    input . add ( ip [ i ] );
  }
  Queue < Integer > output = new LinkedList < > ( );
  for ( int i = 0;
  i < n;
  i ++ ) {
    output . add ( op [ i ] );
  }
  Stack < Integer > tempStack = new Stack < > ( );
  while ( ! input . isEmpty ( ) ) {
    int ele = input . poll ( );
    if ( ele == output . peek ( ) ) {
      output . poll ( );
      while ( ! tempStack . isEmpty ( ) ) {
        if ( tempStack . peek ( ) == output . peek ( ) ) {
          tempStack . pop ( );
          output . poll ( );
        }
        else break;
      }
    }
    else {
      tempStack . push ( ele );
    }
  }
  return ( input . isEmpty ( ) && tempStack . isEmpty ( ) );
}

File: results/py/STACK_PERMUTATIONS_CHECK_IF_AN_ARRAY_IS_STACK_PERMUTATION_OF_OTHER.cpp
bool f_filled ( int * ip , int * op , int n ) {
  queue < int > Input ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) Input . push ( ip [ i ] ) ;
  queue < int > Output ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) Output . push ( op [ i ] ) ;
  vector < int > TempStack ;
  while ( ( ! Input . empty ( ) ) && ( ( Input . front ( ) == Output . front ( ) ) || ( Input . front ( ) == Output . back ( ) ) ) ) {
    int ele = Input . front ( ) ;
    Input . pop ( ) ;
    if ( ( ele == Output . front ( ) ) && ( ( TempStack . size ( ) != 0 ) || ( TempStack . size ( ) == 0 ) ) {
      if ( ( TempStack . back ( ) == Output . front ( ) ) || ( TempStack . back ( ) == 0 ) ) {
        TempStack . pop_back ( ) ;
        Output . pop ( ) ;
      }
      else break ;
    }
  }
  return ( Input . empty ( ) && ( TempStack . size ( ) == 0 ) ) ;
}
